modules/xmpp/xmpp.js

```diff
@@ -57,6 +59,45 @@ function XMPP(options, token) {
 }
 
 /**
+ * Reloads the XMPP module
+ * @param options {object} options to be overriden
+ */
+XMPP.prototype.reload = function (options) {
+    if(!options)
+        options = {};
+    if(!this.options) {
+        this.options = options;
+    } else {
+        // Override config options
```

Comment:
That's the product of copy+paste! The comment is incorrect because there's no "config options". Anyway, the bigger concern is that 9 or 10 lines of source code are repeated for what may (if the thing with options.config in my first comment is not a real requirement) turn out to be 1 function which could be optimized & reused.